archunit/src/test/java/com/tngtech/archunit/library/modules/syntax/ModulesShouldTest.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    @Test
+    public void respectTheirAllowedDependenciesDeclaredIn_takes_allowed_dependencies_from_annotation_property() {
+        assertThatRule(modules().definedByAnnotation(TestModule.class)
+                .should().respectTheirAllowedDependenciesDeclaredIn("allowedDependencies", consideringOnlyDependenciesBetweenModules()))
+                .checking(new ClassFileImporter().importPackagesOf(ClassInModule1.class, InternalClassInModule2.class, ClassInModule3.class))
+                .hasViolationContaining(ClassInModule3.class.getName())
+                .hasNoViolationContaining(ApiClassInModule2.class.getName())
+                .hasNoViolationContaining(InternalClassInModule2.class.getName());
+    }
+
+    @Test
+    public void respectTheirAllowedDependenciesDeclaredIn_works_together_with_filtering_by_predicate() {
+        assertThatRule(modules().definedByAnnotation(TestModule.class)
+                .that(DescribedPredicate.describe("are not Module 1", it -> !it.getName().equals("Module 1")))
+                .and(DescribedPredicate.describe("are not Module 2", it -> !it.getName().equals("Module 2")))
+                .or(DescribedPredicate.describe("are Module 3", it -> it.getName().equals("Module 3")))
+                .should().respectTheirAllowedDependenciesDeclaredIn("allowedDependencies", consideringOnlyDependenciesBetweenModules()))
+                .checking(new ClassFileImporter().importPackagesOf(ClassInModule1.class, InternalClassInModule2.class, ClassInModule3.class))
+                .hasNoViolation();
+    }
+
+    @Test
+    public void respectTheirAllowedDependenciesDeclaredIn_rejects_missing_property() {
+        assertThatThrownBy(
+                () -> modules().definedByAnnotation(TestModule.class)
+                        .should().respectTheirAllowedDependenciesDeclaredIn("notThere", consideringOnlyDependenciesBetweenModules())
+                        .evaluate(new ClassFileImporter().importPackagesOf(ClassInModule1.class, InternalClassInModule2.class))
+        )
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(String.format("Could not invoke @%s.notThere()", TestModule.class.getSimpleName()));
+    }
+
+    @Test
+    public void respectTheirAllowedDependenciesDeclaredIn_rejects_property_of_wrong_type() {
+        assertThatThrownBy(
+                () -> modules().definedByAnnotation(TestModule.class)
+                        .should().respectTheirAllowedDependenciesDeclaredIn("name", consideringOnlyDependenciesBetweenModules())
+                        .evaluate(new ClassFileImporter().importPackagesOf(ClassInModule1.class, InternalClassInModule2.class))
+        )
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(String.format("Property @%s.name() must be of type %s[]", TestModule.class.getSimpleName(), String.class.getSimpleName()));
```

Comment:
I would either use
```suggestion
                .hasMessageContaining(String.format("Property @%s.name() must be of type %s", TestModule.class.getSimpleName(), String[].class.getSimpleName()));
```
or directly inline it
```suggestion
                .hasMessageContaining(String.format("Property @%s.name() must be of type String[]", TestModule.class.getSimpleName()));
```

Reply:
Same in `onlyDependOnEachOtherThroughPackagesDeclaredIn_rejects_property_of_wrong_type`.